SPEC.md

```diff
@@ -0,0 +1,97 @@
+
+### Load testing
+
+These tests are intended to verify the node remains healthy under "reasonable load". Additionally these tests will be pushed to the extreme for resistance testing with heavier loads.
+
+### Heavy load testing
+
+These tests are meant to explore the impact of malicious network use against a node.
+
+The amount of load and its frequency could be modulated to provide a comprehensive verification of the node's behaviour under different conditions (including synchronized requests from different peers and other worst case scenarios).
+
+## Resistance
+
+The resistance tests are designed for the early detection and avoidance of weaknesses exploitable through malicious behaviour. They attempt to probe boundary conditions with comprehensive fuzz testing and extreme load testing. The nature of the peers in these cases will depend on how accurately they needs to simulate node behaviour. It will likely be a mixture of simple sockets for the simple cases and peers used in the performance tests for the more advanced.
+
+### Fuzz testing
+
+The fuzz tests aim to buttress the message conformance tests with extra verification of expected node behaviour when receiving corrupted or broken messages. Our approach is targeting these specific areas and we anticipate broadening these test scenarios as necessary:
+
+- Messages with any length and any content (random bytes).
+- Messages with plausible lengths, e.g. 24 bytes for header and within the expected range for the body.
+- Metadata-compliant messages, e.g. correct header, random body.
+- Slightly corrupted but otherwise valid messages, e.g. N% of body replaced with random bytes.
+- Messages with an incorrect checksum.
+- Messages with differing announced and actual lengths.
+
+# Test Index
+
+The test index makes use of symbolic language in describing connection and message sending directions. As a convention, Ziggurat test nodes are to the left of the connection/message arrows, and Algod instances are to the right: `A -> B` and `A <- B`. In this way, `->` signifies "Ziggurat connects to Algod" and `<-` signifies the opposite. Furthermore, `-> ping` signifies "Ziggurat sends a `Ping` message to Algod" and `<- pong` signifies "Algod sends a `Pong` message to Ziggurat". Lastly, `<>` signifies a completed handshake, in either direction.
+
+## Conformance
+
+### ZG-CONFORMANCE-001
+
+    The node correctly performs a handshake from the responder side.
+
+    ->
+    -> http handshake request (websocket upgrade)
+    <- http handshake response (websocket upgrade accept)
+
+    Assert: the node’s peer count has increased to 1 and the synthetic node is an established peer.
```

Comment:
```
    -> http handshake request (websocket upgrade)
    <- http handshake response (websocket upgrade accept)
```
 I would gladly accept a better solution here 🙂

Reply:
I think this is quite clear, what would you like to improve?

Reply:
> I think this is quite clear

Then it's all good. I wasn't sure if that was the fanciest way to show an HTTP handshake.